src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -659,44 +605,47 @@ program_plp(const xrt_core::device* dev, const std::string& partition)
   std::cout << "Programmed shell successfully" << std::endl;
 }
 
+/**
+ * @brief Create a list of images that are known to exist based on given names and paths
+ * 
+ * @param[in] image_list The list of images paths and names to be validated
+ * @return std::vector<std::string> A list of known and valid images for flashing
+ */
 static std::vector<std::string>
 find_flash_image_paths(const std::vector<std::string> image_list)
 {
   std::vector<std::string> path_list;
   auto installedShells = firmwareImage::getIntalledDSAs();
   
-  //iterates over installed shells
-  //checks the vbnv against the vnbv passed in by the user
-  auto get_shell_file = [installedShells] (std::string shell_name) {
-    std::string path;
-    int multiple_shells = 0;
-    
-    for(auto const& shell : installedShells) {
-      if(shell_name.compare(shell.name) == 0) {
-        multiple_shells++;
-        path = shell.file;
-      }
-    }
+  for(const auto& img : image_list) {
+    // Check if the passed in image is absolute path
+    if(boost::filesystem::exists(boost::filesystem::path(img)))
+      path_list.push_back(img);
+    // Search installed shells and find the image path by checking the vbnv
+    else {
 
-    //error-handling
-    if(multiple_shells == 0) 
-      throw xrt_core::error("Specified base not found on the system");
-    
-    //if multiple shells with the same vbnv are installed on the system, we don't want to 
-    //blindly update the device. in this case, the user needs to specify the complete path
-    if(multiple_shells > 1) 
-      throw xrt_core::error("Specified base matched mutiple installed bases. Please specify the full path.");
+      /*
+       * Iterate over installed shells
+       * Checks the vbnv against the vnbv passed in by the user
+       */
+      std::string img_path = "";
```

Comment:
Do not repeat what the constructor does
```suggestion
      std::string img_path;
```